From patchwork Fri Sep 21 22:44:23 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jim Mattson <jmattson@google.com>
X-Patchwork-Id: 10611165
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7FA4B15E8
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 21 Sep 2018 22:44:32 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 689672DDB8
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 21 Sep 2018 22:44:32 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 5BC8F2DDC8; Fri, 21 Sep 2018 22:44:32 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.5 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	USER_IN_DEF_DKIM_WL autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 007C82DDB8
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 21 Sep 2018 22:44:31 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726011AbeIVEf1 (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Sat, 22 Sep 2018 00:35:27 -0400
Received: from mail-ua1-f73.google.com ([209.85.222.73]:35589 "EHLO
        mail-ua1-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725750AbeIVEf1 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Sat, 22 Sep 2018 00:35:27 -0400
Received: by mail-ua1-f73.google.com with SMTP id l22-v6so3676363uak.2
        for <kvm@vger.kernel.org>; Fri, 21 Sep 2018 15:44:29 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:message-id:mime-version:subject:from:to:cc;
        bh=YKY8CS0ortvrR+QrDKQMotWrMG6BcnUKAr/Iv+ER9t0=;
        b=JAGNNxBCtHx9LUtqFiOnSPzzXIygn8wdXhwFZficDA8+wbCpZDzPgxxswIzyuWGXQ/
         Yo5u4Dd9l4LYo6pm25OVuDL/S0Qhf07UERZaxu3nwRSsjsoKf0x2xw2nXlH5P4GLa2ZX
         BJR5hTS/UFO99bzcskcmqivhP2fqetdj+d/oRNvaS2TzGidP/MPD7o/ZfMWlEWD07SlI
         vt0t0pAHRkRkbhsGRPuq8xYI84o6ccOYtr42Hao/amFc8MFT73mUMoF86Cbw/4vS2iAa
         oig6Lx2xcCrFTjHmt8Qw9ZGEJEozdf9BWXKvOs0CVPQrAbHTYHmF8OlYNXRFQtj+bDGF
         xsTg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:message-id:mime-version:subject:from:to:cc;
        bh=YKY8CS0ortvrR+QrDKQMotWrMG6BcnUKAr/Iv+ER9t0=;
        b=NW8LIyrLXTI4iYq7iFtaENj8x0KkPOHvwoM/c5xQdhnTrT1uIxxWYrmHEhPSIci1+I
         kzDLLKYU5PNjMFm+x3RIaL2V8pdODS3K7o7OeV3F32xAGL5f5hWDZoWIad7Fj5d5rMd4
         5nxf3RidzmkLvPi/1D0QqJlJEM1VYoTJxTbbNQzhggCgsC+f83qA+p/7LE1RFkT4Lgmi
         zfFi89q0qvbOV9oeS/9BmVgfVM4OupnalGkRtIUdlrqzepC+dc7GGt+UmzS9sg1ZeNS0
         f4tZ1SjPAVgyGJSOXFkMQgfloqyL/eJomwurp8AQSu9Fz5RfWudCbCPZYPLL2u8DjA9Z
         XewQ==
X-Gm-Message-State: APzg51BF1hg8fJl5fu9B9vcE7/UsZ1Esoq9x+d1fxH7YxX2oChMa9lMV
        2t/UF22gTmxicYd0uUeGRUQvPSzQUTaIn3a5j96Iq6qneQuOSpwFQzd1kEVsu+O91C27Wk3H74V
        watysqIr0GKbe0dWG9zVWUCgtUK/ZDZOW1f5z3zGxJjYYoio+hSnD8h6H+jPABjc=
X-Google-Smtp-Source: 
 ANB0VdbK1REYYGOTLgTMOx2jPRpTrW0jG1AH8sA5D9/MV8b6kpYr6To7kAR7bOJ6seY778i1EiS4TeMT14vtOg==
X-Received: by 2002:ab0:4a01:: with SMTP id
 q1-v6mr8811498uae.18.1537569869174;
 Fri, 21 Sep 2018 15:44:29 -0700 (PDT)
Date: Fri, 21 Sep 2018 15:44:23 -0700
Message-Id: <20180921224423.105568-1-jmattson@google.com>
Mime-Version: 1.0
X-Mailer: git-send-email 2.19.0.444.g18242da7ef-goog
Subject: [PATCH] KVM: x86: Remove documentation for KVM_CAP_INTR_SHADOW
From: Jim Mattson <jmattson@google.com>
To: kvm@vger.kernel.org
Cc: Jim Mattson <jmattson@google.com>
Content-Type: text/plain; charset="UTF-8"
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This capability was never implemented. Aside from the #define, there
are no references to it anywhere in the code base. Let's remove the
misleading documentation.

Signed-off-by: Jim Mattson <jmattson@google.com>
---
 Documentation/virtual/kvm/api.txt | 4 +---
 include/uapi/linux/kvm.h          | 2 +-
 tools/include/uapi/linux/kvm.h    | 2 +-
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/Documentation/virtual/kvm/api.txt b/Documentation/virtual/kvm/api.txt
index 647f94128a85..e15aaa460915 100644
--- a/Documentation/virtual/kvm/api.txt
+++ b/Documentation/virtual/kvm/api.txt
@@ -834,7 +834,6 @@ struct kvm_clock_data {
 4.31 KVM_GET_VCPU_EVENTS
 
 Capability: KVM_CAP_VCPU_EVENTS
-Extended by: KVM_CAP_INTR_SHADOW
 Architectures: x86, arm, arm64
 Type: vcpu ioctl
 Parameters: struct kvm_vcpu_event (out)
@@ -933,7 +932,6 @@ struct kvm_vcpu_events {
 4.32 KVM_SET_VCPU_EVENTS
 
 Capability: KVM_CAP_VCPU_EVENTS
-Extended by: KVM_CAP_INTR_SHADOW
 Architectures: x86, arm, arm64
 Type: vcpu ioctl
 Parameters: struct kvm_vcpu_event (in)
@@ -955,7 +953,7 @@ KVM_VCPUEVENT_VALID_NMI_PENDING - transfer nmi.pending to the kernel
 KVM_VCPUEVENT_VALID_SIPI_VECTOR - transfer sipi_vector
 KVM_VCPUEVENT_VALID_SMM         - transfer the smi sub-struct.
 
-If KVM_CAP_INTR_SHADOW is available, KVM_VCPUEVENT_VALID_SHADOW can be set in
+Beginning with the 2.6.35 kernel, KVM_VCPUEVENT_VALID_SHADOW can be set in
 the flags field to signal that interrupt.shadow contains a valid state and
 shall be written into the VCPU.
 
diff --git a/include/uapi/linux/kvm.h b/include/uapi/linux/kvm.h
index 251be353f950..7fcbe0e60ec7 100644
--- a/include/uapi/linux/kvm.h
+++ b/include/uapi/linux/kvm.h
@@ -836,7 +836,7 @@ struct kvm_ppc_resize_hpt {
 #define KVM_CAP_HYPERV_SPIN 46
 #define KVM_CAP_PCI_SEGMENT 47
 #define KVM_CAP_PPC_PAIRED_SINGLES 48
-#define KVM_CAP_INTR_SHADOW 49
+#define KVM_CAP_INTR_SHADOW 49 /* never implemented */
 #ifdef __KVM_HAVE_DEBUGREGS
 #define KVM_CAP_DEBUGREGS 50
 #endif
diff --git a/tools/include/uapi/linux/kvm.h b/tools/include/uapi/linux/kvm.h
index 07548de5c988..de03dfad207f 100644
--- a/tools/include/uapi/linux/kvm.h
+++ b/tools/include/uapi/linux/kvm.h
@@ -836,7 +836,7 @@ struct kvm_ppc_resize_hpt {
 #define KVM_CAP_HYPERV_SPIN 46
 #define KVM_CAP_PCI_SEGMENT 47
 #define KVM_CAP_PPC_PAIRED_SINGLES 48
-#define KVM_CAP_INTR_SHADOW 49
+#define KVM_CAP_INTR_SHADOW 49 /* never implemented */
 #ifdef __KVM_HAVE_DEBUGREGS
 #define KVM_CAP_DEBUGREGS 50
 #endif
